FairyTaleDefender/Assets/_Game/Scripts/Tests/Runtime/Systems/SaveGameSystem/ScriptableObjects/SaveGameManagerSOTests.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections;
+using System.IO;
+using BoundfoxStudios.FairyTaleDefender.Common;
+using BoundfoxStudios.FairyTaleDefender.Extensions;
+using BoundfoxStudios.FairyTaleDefender.Infrastructure.FileManagement;
+using BoundfoxStudios.FairyTaleDefender.Systems.SaveGameSystem;
+using BoundfoxStudios.FairyTaleDefender.Systems.SaveGameSystem.ScriptableObjects;
+using Cysharp.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace BoundfoxStudios.FairyTaleDefender.Tests.Systems.SaveGameSystem.ScriptableObjects
+{
+	public class SaveGameManagerSOTests
+	{
+		private const string BackupDirectoryName = "Save Games Backup";
+
+		private string SaveGamesPath => Path.Combine(Application.persistentDataPath, Constants.SaveGames.DirectoryName);
+		private string SaveGamesBackupPath => Path.Combine(Application.persistentDataPath, BackupDirectoryName);
+
+		[SetUp]
+		public void BackupSaveGamesFolder()
+		{
+			if (Directory.Exists(SaveGamesPath))
+			{
+				Directory.Move(SaveGamesPath, SaveGamesBackupPath);
+			}
+		}
+
+		[TearDown]
+		public void RestoreSaveGamesFolderBackup()
+		{
+			if (Directory.Exists(SaveGamesBackupPath))
+			{
+				if (Directory.Exists(SaveGamesPath))
+				{
+					Directory.Delete(SaveGamesPath, true);
+				}
+
+				Directory.Move(SaveGamesBackupPath, SaveGamesPath);
+			}
```

Comment:
Invert if/ Guard Clause draus machen?
```suggestion
	if (!Directory.Exists(SaveGamesBackupPath)) return;
	if (Directory.Exists(SaveGamesPath))
	{
		Directory.Delete(SaveGamesPath, true);
	}
	Directory.Move(SaveGamesBackupPath, SaveGamesPath);
```